ecl/eclcmd/eclcmd_core.cpp

```diff
@@ -1959,15 +1964,39 @@ class EclCmdZapGen : public EclCmdCommon
             if ((strstr(errMsg.str(), "Warning") != nullptr) || (strstr(errMsg.str(), "Failed") != nullptr))
             {
                 fprintf(stderr, "Zapgen command returns with error:\n%s\n", errMsg.str());
-                return false;
+                isOk = false;
+                removeOutFile = true; // It contains the authentication error message only
             }
-            else if (optVerbose)
+            else if (strstr(errMsg.str(), "Authentication problem") != nullptr)
+            {
+                fprintf(stderr, "\nZapgen command returned with an authentication problem. Please check your credentials.\n");
+                isOk = false;
+                removeOutFile = true; // It contains the authentication error message only
+            }
+            else if (strstr(errMsg.str(), "401 Unauthorized") != nullptr)
+            {
+                fprintf(stderr, "Zapgen command needs to authenticate, please provide credentials.\n");
+                isOk = false;
+                removeOutFile = true; // It contains the authentication error message only
+            }
+
+            if (optVerbose)
                 fprintf(stderr, "Zapgen command returns with:\n%s\n", errMsg.str());
```

Comment:
@AttilaVamos picky and not because of this PR, but should be "returned".

Reply:
Changed